Remove CarbonIcons package and use SVGs for all icons

untreu2 opened the pull request:
<!--
  Thanks for contributing!

  Provide a description of your changes below and a general summary in the title

  Please look at the following checklist to ensure that your PR can be accepted quickly:
-->

## Description
Removes the CarbonIcons package and implements the direct use of SVG files from Figma for all icons. This change streamlines icon management and improves the workflow between design and development.

<!--- Describe your changes in detail -->

## Type of Change

<!--- Put an `x` in all the boxes that apply: -->

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

## Checklist

<!-- Please make sure you've done the following before submitting your PR: -->

- [x] Run `just precommit` to ensure that formatting and linting are correct
- [ ] Updated the `CHANGELOG.md` file with your changes

Fixes #474 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Style**
  * Migrated app-wide icons to tinted SVG assets for crisper, consistent visuals across chat, contacts, reactions, settings, and headers.

* **Refactor**
  * Standardized asset-based icon handling and status indicator rendering throughout the UI.

* **Tests**
  * Updated UI tests to assert SVG-based icons.

* **Chores**
  * Removed legacy icon-font dependency and related assets; deleted in-chat audio playback component (audio playback UI and controls removed).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

josefinalliende commented on lib/ui/contact_list/chat_list_screen.dart:
```diff
@@ -370,9 +370,10 @@ class _ChatListScreenState extends ConsumerState<ChatListScreen> with TickerProv
-                                    CarbonIcons.close,
-                                    size: 24.w,
+                                  child: SvgPicture.asset(
+                                    AssetsPaths.icClose,
```
Related to previous comment, there is an issue on this one

josefinalliende commented on lib/ui/contact_list/new_chat_bottom_sheet.dart:
```diff
@@ -383,10 +382,14 @@ class _NewChatBottomSheetState extends ConsumerState<NewChatBottomSheet> {
-              color: context.colors.primary,
-              size: 20.sp,
+            prefixIcon: SvgPicture.asset(
+              AssetsPaths.icSearch,
```
Probably related to the previosu comment cause it is the same icon. Here the issue is only size, not color. This is how it looks
|Light mode| Dark mode|
|----|----|
|<img width="363" height="758" alt="Screenshot 2025-08-16 at 12 00 26 AM" src="https://github.com/user-attachments/assets/3deaf6ed-0e81-4e11-91bc-425f547e6ab4" />|<img width="371" height="762" alt="Screenshot 2025-08-16 at 12 01 28 AM" src="https://github.com/user-attachments/assets/a53436df-4b88-41c8-9a5f-442f3fc1bf72" />|

josefinalliende commented on lib/ui/core/ui/wn_heads_up.dart:
```diff
@@ -1,6 +1,7 @@
+import 'package:whitenoise/ui/core/themes/assets.dart';
 import 'package:whitenoise/ui/core/themes/src/app_theme.dart';
 
 class WnStickyHeadsUp extends StatelessWidget {
```
Oopps probably my bad here when adding Wn prefix to existing components, but this file name does not match the widget name. Can be fixed in other issue, just commenting it here to not forget to fix it.

josefinalliende commented on lib/ui/settings/network/widgets/relay_tile.dart:
There is an issue here with colors for statuses, now I don't see them in green. now. In light mode happens the same


<img width="381" height="764" alt="Screenshot 2025-08-16 at 12 28 33 AM" src="https://github.com/user-attachments/assets/e9cac459-1c40-4c19-9082-138aff5ccc52" />

josefinalliende reviewed (changes requested):
Great job! 🚀 🎉   Thank you for doing this one!!  Reviewed it locally and overall it is working really well!  

💡 I think that the idea  mentioned in the signal chat of having an icon component is great, but definitely would leave that for other issue. It could be mayb e a subissue of the issue you are solving now (and someone else can take it, I am not saying thise for you to take it if you maybe need/have/want to do something else now).  If you are ok with this just let me know and I can create the sub-issue.

Last but not least, I caught a little bug in some sizes and colors, left screenshots of the problems I found. 

<img width="250"  alt="caught-little-bug" src="https://github.com/user-attachments/assets/c70b8b62-2ec8-412a-b5eb-19a3f0e2afc6" />

untreu2 commented:
@josefinalliende I fixed the sizing/coloring bugs and removed audio-related files.

<img width="521" height="899" alt="image" src="https://github.com/user-attachments/assets/f5ad5a2d-8f05-492b-8547-edb041ca53cd" />

<img width="521" height="899" alt="image" src="https://github.com/user-attachments/assets/ff4a72e7-8d1d-41d8-9ccc-9cddbeec4394" />

<img width="521" height="899" alt="image" src="https://github.com/user-attachments/assets/1ec5aeb8-4676-40b7-9432-a3888198634b" />

josefinalliende commented on lib/ui/chat/widgets/chat_contact_avatar.dart:
Opened issue #489
